src/science/canopy/cbl_zetar.F90

```diff
@@ -13,34 +15,85 @@ SUBROUTINE update_zetar( mp, NITER, canopy_zetar, iter, nrb, CVONK, CGRAV, CCAPP
                    cable_user_soil_struc, air_rho, met_tk,  met_fsd, &
                    rough_zref_tq, rough_hruff, rough_term6a, rough_z0soilsn,   &
                    canopy_vlaiw, canopy_zetash,  canopy_us, &
-                   canopy_fh, canopy_fe, canopy_fhs, canopy_fes ) 
+                   canopy_fh, canopy_fe, canopy_fhs, canopy_fes )
+  !* This SUBROUTINE forms part of the codebase to evaluate the surface
+  !  energy balance on a sub-diurnal basis (i.e. every CABLE time step).
+  !  It resides in the canopy science directory and is called from
+  !  [[define_canopy]] only.
+  !  
+  !  This SUBROUTINE updates the value of the stability parameter \(\xi\)
+  !  during the Monin-Obukhov (MO) iteration loop in [[define_canopy]].
+  !  Further scientific documentation is given in
+  !  [Kowalczyk et al. (2006)](http://www.cmar.csiro.au/e-print/open/kowalczykea_2006a.pdf)
+  !  - section 3.1, equations 1-9.  The two outputs of the SUBROUTINE
+  !  are the local (in space, time and by iteration counter) value of \(\xi\)
+  !  (Equation 9) `canopy_zetar` and a related quantity `canopy_zetash`.
+  !
+  !  `canopy_zetar` is evaluated from the total land (soil+canopy)
+  !  surface fluxes of momentum, sensible heat and latent heat.
+  !  `canopy_zetash` is the equivalent variable evaluated from the soil
+  !  contribution to those fluxes only.  `canopy_zetash` is used in conjunction
+  !  with the [[sli_main_mod]] soil model to moderate the fluxes from the soil
+  !  underneath a canopy.
+  !
+  !  `canopy_zetar` and `canopy_zetash` quantify the role that the surface
+  !  fluxes play in setting the efficiency of turbulent transfer from the land
+  !  to the atmosphere, and hence the aerodynamic component of the resistance
+  !  network for those same surface fluxes (an implicit problem which requires
+  !  iteration to solve).
+  !
+  !  <br></br>
+  !
+  !  `canopy_zetar` and `canopy_zetash` are initialised to `CZETA0`=0 in
+  !  [[define_canopy]] and updated `NITER`(>1) times
+  !  during the calculation of the energy balance.  The value of the variables
+  !  at each iteration are stored in memory to aid in the diagnosis
+  !  of convergence.
+  !
+  !  The outputs `canopy_zetar` and `canopy_zetash` are known as
+  !  `canopy%zetar` and `canopy%zetash` in [[define_canopy]] and elsewhere
+  !  in the code. `canopy%zetar` is used in [[define_canopy]] and
+  !  [[comp_friction_vel]]; `canopy%zetash` is used in `[[define_canopy]].
+  !
+  !  Special cases apply if `NITER`=2, or if `canopy_zetar` or `canopy_zetash`
+  !  exceed the prescribed upper `CZETPOS` or lower `CZETNEG` limits. `NITER`(=4)
+  !  is defined in [[cable_types_mod]]; `CZETMUL`, `CZET0`, `CZETPOS` and
+  !  `CZETNEG` in [[cable_phys_constants_mod]].
```

Comment:
```suggestion
  !  A special case applies if `NITER`=2.`canopy_zetar` and `canopy_zetash`
  !  are also bounded by the interval `[CZETNEG, CZETPOS]`.
  !
  !`NITER`(=4) is defined in [[cable_types_mod]]; `CZETMUL`, `CZET0`, `CZETPOS` and
  !  `CZETNEG` in [[cable_phys_constants_mod]].
```
I am not sure I am too keen on the end here that tells where the constants come from.

Reply:
Again this was included to aid navigation - at the moment a reader has no idea where these are set (namelists, internal variables, or hard-wired in the code)

Reply:
Left in for now